docs/dev/background-information/bazel/faq.mdx

```diff
@@ -348,6 +348,42 @@ rust_binary(
 )
 ```
 
+### `bazel sync` authentication failure when cloning `syntect`
+
+When repinning dependencies with `CARGO_BAZEL_REPIN=1 CARGO_BAZEL_REPIN_ONLY=crate_index bazel sync --only=crate_index` it may fail with the following Cargo error:
+
+```
+STDERR ------------------------------------------------------------------------
+
+    Updating crates.io index
+    Updating git repository `https://github.com/sourcegraph/syntect`
+error: failed to get `syntect` as a dependency of package `sg-syntax v0.1.0 (/var/folders/1r/5z42n9p52zv8rfp93gxc1vfr0000gn/T/.tmpyhlucq/crates/sg-syntax)`
+
+Caused by:
+  failed to load source for dependency `syntect`
+
+Caused by:
+  Unable to update https://github.com/sourcegraph/syntect?rev=7e02c5b4085e6d935b960b8106cdd85da04532d2#7e02c5b4
+
+Caused by:
+  failed to clone into: /private/var/tmp/_bazel_william/c92ec739369034d3064b6df55c419545/external/crate_index/.cargo_home/git/db/syntect-383b2f29eb0ef0d0
+
+Caused by:
+  failed to authenticate when downloading repository: ssh://git@github.com/sourcegraph/syntect
+
+  * attempted ssh-agent authentication, but no usernames succeeded: `git`
+
+  if the git CLI succeeds then `net.git-fetch-with-cli` may help here
+  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
+
+Caused by:
+  no authentication methods succeeded
+
+Error: Failed to update lockfile: exit status: 101
+```
+
+You might be able to git ssh clone that repository locally yet in Bazel Cargo it fails. This is because Bazel Cargo doesn't use your `~/.ssh/config` file and thus can't use your SSH private key. The error says you can set `net.git-fetch-with-cli` in `Cargo.toml` or configure a credential helper. All cargo settings also have environment variable variants, so you can do the repinning with `CARGO_NET_GIT_FETCH_WITH_CLI=true CARGO_BAZEL_REPIN=1 bazel sync --only=crate_index` without setting the value in the `Cargo.toml`.
```

Comment:
```suggestion
You might be able to Git SSH clone that repository locally, yet in Bazel Cargo, it fails. This is because Bazel Cargo doesn't use your `~/.ssh/config` file and thus can't use your SSH private key. The error says you can set `net.git-fetch-with-cli` in `Cargo.toml` or configure a credential helper. All cargo settings also have environment variable variants, so you can do the repinning with `CARGO_NET_GIT_FETCH_WITH_CLI=true CARGO_BAZEL_REPIN=1 bazel sync --only=crate_index` without setting the value in the `Cargo.toml`.
```